### PR TITLE
1970 canary date component playgrounds

### PR DESCRIPTION
**Pull request by GCHQ-Developer-530**
## Summary of the changes
Add playground stories for date input and date picker

Ticket already created for startOfWeek not updating - https://github.com/mi6/ic-ui-kit/issues/1877

## Related issue
Part of #1970

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.

### CODE REVIEW

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Review by ad9242 — CHANGES_REQUESTED** [review]
i think it would be quite nice to make value a string, then we could test with "-" separator too

**Comment by ad9242** [comment]
There seems to be a problem where, if you have a value set, changing the date-format clears the field. I'd expect it to update the value, but that should be a new ticket as it doesn't support it at the moment (would need changes to input & picker i think)

**Comment by GCHQ-Developer-530** [comment]
> There seems to be a problem where, if you have a value set, changing the date-format clears the field. I'd expect it to update the value, but that should be a new ticket as it doesn't support it at the moment (would need changes to input & picker i think)

https://github.com/mi6/ic-ui-kit/issues/2056 created

**Review by ad9242 — CHANGES_REQUESTED** [review]
I think the blanking of the value may be a bigger issue, than just changing the date format. it seems to be that, if you have entered a value in the picker\input and then change a storybook control eg, disabled, then it blanks the value. I think this is because it re-uses the value from the storybook control.

you can keep them in sync with https://storybook.js.org/docs/addons/addons-api#useargs 
see checkbox playground for an example

**Comment by GCHQ-Developer-530** [comment]
> I think the blanking of the value may be a bigger issue, than just changing the date format. it seems to be that, if you have entered a value in the picker\input and then change a storybook control eg, disabled, then it blanks the value. I think this is because it re-uses the value from the storybook control.
> 
> you can keep them in sync with https://storybook.js.org/docs/addons/addons-api#useargs see checkbox playground for an example

Ah brilliant, thank you. I've updated it

File: packages/canary-react/src/stories/ic-date-input.stories.mdx [hunk 1]
```diff
@@ -607,3 +608,86 @@ The date input can be cleared by setting the the value attribute to one of the f
...
+          openAtDate={args.openAtDate}
+          required={args.required}
+          showDaysOutsideMonth={args.showDaysOutsideMonth}
+          showPickerClearButton={args.showPickerClearButton}
+          showPickerTodayButton={args.showPickerTodayButton}
+          size={args.size}
+          startOfWeek={args.startOfWeek}
+          validationStatus={args.validationStatus}
+          validationText={args.validationText}
+          value={value}
```
Comment (ad9242) [inline packages/canary-react/src/stories/ic-date-input.stories.mdx:689]: think you need to add an onIcChange event that calls updateArgs with the new value, to keep the control in sync:

```
onIcCheck={(ev) => updateArgs({ value: ev.detail.value })}
```

**Review by ad9242 — CHANGES_REQUESTED** [review]

File: packages/canary-react/src/stories/ic-date-picker.stories.mdx [hunk 2]
```diff
@@ -506,4 +508,109 @@ const DatePickerWithIcChange = () => {
...
+    }}
+    name="Playground example"
+  >
+    {(args) => {
+      const [{ value }, updateArgs] = useArgs();
+
+      const updateDate = (ev) => {
+        const date = ev.detail.value;
+        const formattedDate = `${date.getDate()}-${date.getMonth() + 1}-${date.getFullYear()}`;
+        updateArgs({ value: formattedDate});
```
Comment (ad9242) [inline packages/canary-react/src/stories/ic-date-picker.stories.mdx:584]: sorry, could this also handle setting an empty string when the value is cleared?

**Review by ad9242 — CHANGES_REQUESTED** [review]

**Review by ad9242 — APPROVED** [review]

**Review by GCHQ-Developer-847 — APPROVED** [review]

**Pull request merged**